lib/boombox/internal_bin.ex

```diff
@@ -0,0 +1,591 @@
+defmodule Boombox.InternalBin do
+  @moduledoc """
+  TODO: write docs
+  """
+  use Membrane.Bin
+
+  require Membrane.Logger
+  require Membrane.Transcoder.Audio
+  require Membrane.Transcoder.Video
+  alias Membrane.Transcoder
+
+  @type input() ::
+          (path_or_uri :: String.t())
+          | {:mp4, location :: String.t(), transport: :file | :http}
+          | {:webrtc, Boombox.webrtc_signaling()}
+          | {:whip, uri :: String.t(), token: String.t()}
+          | {:rtmp, (uri :: String.t()) | (client_handler :: pid)}
+          | {:rtsp, url :: String.t()}
+          | {:rtp, Boombox.in_rtp_opts()}
+          | {:stream, pid(), Boombox.in_stream_opts()}
+          | :membrane_pad
+
+  @type output ::
+          (path_or_uri :: String.t())
+          | {path_or_uri :: String.t(), [Boombox.force_transcoding()]}
+          | {:mp4, location :: String.t()}
+          | {:mp4, location :: String.t(), [Boombox.force_transcoding()]}
+          | {:webrtc, Boombox.webrtc_signaling()}
+          | {:webrtc, Boombox.webrtc_signaling(), [Boombox.force_transcoding()]}
+          | {:whip, uri :: String.t(), [{:token, String.t()} | {bandit_option :: atom(), term()}]}
+          | {:hls, location :: String.t()}
+          | {:hls, location :: String.t(), [Boombox.force_transcoding()]}
+          | {:rtp, Boombox.out_rtp_opts()}
+          | {:stream, pid(), Boombox.out_stream_opts()}
```

Comment:
I'd avoid duplicating these types. The easiest way would be `input() :: Boombox.input() | {:stream, pid(), Boombox.in_stream_opts()}`, preferably I'd pass the PID via a separate option